### PR TITLE
Backend optimization

### PR DESCRIPTION
**Pull request by elzadj**
# Description

1. Keeps the original breakpoint when the user opens the sidebars in the editor. That reduces a number of re-renders.
2. Adds a scrollbar to the editor when a sidebar is opened.
3. Removes re-render for some blocks when we click them in the List view (mostly buttons).
4. Fixes styles for Tablet / Mobile Gutenberg views in the Editor.

# How Has This Been Tested?

Please check the above points, FSE included. Check frontend's responsive too, just in case.

# Test checklist

**_ Front/Back Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type

**_ Pre-Code Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced editor interface layout and responsiveness, particularly with the sidebar.
	- Improved font management and responsive preview handling in the MaxiBlockComponent.
- **Bug Fixes**
	- Adjusted the minimum width of the editor wrapper based on responsive size settings.
- **Refactor**
	- Streamlined the ResponsiveSelector component by removing complex responsive preview logic.
	- Integrated scrollbar width calculations into the existing resize handling logic.

### CODE REVIEW

**Review by Olekrut — APPROVED** [review]
it works great) 
one question - why we see horizontal scroll even if all sidebars are hidden? is it ok?
image

**Comment by Olekrut** [comment]
> Nope, and I don't see it, so weird. What's your screen size? And is it for ALL patterns, or just this one? And do you see it if you add just a Container with something?

image
i see it even without any container
image

**Comment by elzadj** [comment]
@Olekrut please try now

File: src/extensions/maxi-block/maxiBlockComponent.js [hunk 1]
```diff
@@ -1222,6 +1222,146 @@
 		isBlockStyleChange,
 		iframe
 	) {
+		if (iframe?.contentDocument?.body) {
```
[the remark this answers is not in the document]
Comment (elzadj) [reply src/extensions/maxi-block/maxiBlockComponent.js:1225]: @coderabbitai re-check the current code

File: src/extensions/maxi-block/maxiBlockComponent.js [hunk 2]
```diff
@@ -1222,6 +1222,146 @@
...
+							const rawFrontStyles = document.querySelector(
+								'#maxi-blocks-block-css'
+							);
+
+							if (rawFrontStyles) {
+								iframe.contentDocument.head.appendChild(
+									rawFrontStyles.cloneNode(true)
+								);
+							}
+						}
```
[the remark this answers is not in the document]
Comment (elzadj) [reply src/extensions/maxi-block/maxiBlockComponent.js:1360]: @coderabbitai re-check the current code

**Review by Olekrut — CHANGES_REQUESTED** [review]
When XXL is base > check XL bp
image

**Review by Olekrut — APPROVED** [review]

**Pull request merged**